src/target/riscv/riscv.c

```diff
@@ -877,52 +878,58 @@ static int maybe_add_trigger_t2_t6(struct target *target,
 			if (ret != ERROR_TARGET_RESOURCE_NOT_AVAILABLE)
 				return ret;
 		}
-		LOG_TARGET_DEBUG(target, "trying to setup GE+LT chained match trigger pair");
-		struct trigger_request_info ge_1 = {
-			.tdata1 = fields.common | fields.size.any | fields.chain.enable |
-				fields.match.ge,
-			.tdata2 = trigger->address,
-			.tdata1_ignore_mask = fields.tdata1_ignore_mask
-		};
-		struct trigger_request_info lt_2 = {
-			.tdata1 = fields.common | fields.size.any | fields.chain.disable |
-				fields.match.lt,
-			.tdata2 = trigger->address + trigger->length,
-			.tdata1_ignore_mask = fields.tdata1_ignore_mask
-		};
-		ret = try_setup_chained_match_triggers(target, trigger, ge_1, lt_2);
-		if (ret != ERROR_TARGET_RESOURCE_NOT_AVAILABLE)
-			return ret;
 
-		LOG_TARGET_DEBUG(target, "trying to setup LT+GE chained match trigger pair");
-		struct trigger_request_info lt_1 = {
-			.tdata1 = fields.common | fields.size.any | fields.chain.enable |
-				fields.match.lt,
-			.tdata2 = trigger->address + trigger->length,
-			.tdata1_ignore_mask = fields.tdata1_ignore_mask
-		};
-		struct trigger_request_info ge_2 = {
+		if (r->enable_ge_lt_trigger) {
```

Comment:
@kr-sc , I'd like to request few additional improvements:

1. `enable_ge_lt_trigger` should be renamed to `wp_enable_ge_lt_trigger` . This is to clearly identify that these controls are for watchpoints onlyl.
2. when `trigger->length > 0` we should allow for equality match to be used regardless of this control (otherwise we won't be able to set such wathcpoints)

Reply:
@timsifive , @JanMatCodasip FYI